src/System.IO.FileSystem/src/System/IO/Win32FileStreamAsyncResult.cs

```diff
@@ -207,7 +207,8 @@ internal void CallUserCallback()
                     // Set completedSynchronously to false, since it's on another 
                     // thread, not the main thread.
                     _completedSynchronously = false;
-                    Task.Run(() => CallUserCallbackWorker());
+                    Task.Factory.StartNew(state => ((FileStreamAsyncResult)state).CallUserCallbackWorker(),
```

Comment:
Considering how much more verbose the new code is than the old, this seems like a good place to add a comment mentioning why this pattern was used for this call.